### ISSUE TITLE
Multiple Rule-Sets in Config w/ Their Own Finder Configuration

### ISSUE THREAD

**Issue by ktomk** [FriendsOfPHP/PHP-CS-Fixer#3595]
It would be great to make the current config a composite which can contain one or more config leafs whereas a node contains of:

* risky-allowed
* rules
* finder

This would allow - in a backwards compatible way - to have different rules on different paths.

**Comment by ktomk** [comment]
This could express things similar than #3126 but have it configuration driven.

**Comment by rybakit** [comment]
I ran into a similar problem that I need to apply some fixers to only specific path(s). For example, I wanted the "native_function_invocation" filter to be applied to only the `src` folder, as I don't care about the performance of my tests and example scripts and don't want them to be cluttered with unnecessary `\`s. I worked around that by creating a simple decorator which wraps a fixer and run it only if the paths match (see [.php_cs.dist](https://github.com/rybakit/msgpack.php/blob/863c0940f8a2727fad2413e0330daa23b0c6cf97/.php_cs.dist)).
